### PR TITLE
[Mobile] - Enable font size and line height support and addresses Android issues

### PR DESCRIPTION
**Pull request by geriux**
- Gutenberg mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4519
- Aztec Android PR: https://github.com/wordpress-mobile/AztecEditor-Android/pull/952
- WordPress Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/15868
- WordPress iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17833

## Description
This PR enables font size and line-height support on mobile for block-based themes.

It also includes some fixes on Android and the Heading block.

On Aztec Android, different scales are added to the Heading block. Unfortunately, I couldn't find an easy way to disable this for an specific case like block-based themes. So for now the approach is the following:

- [Bring the scales](https://github.com/WordPress/gutenberg/pull/38205/files#diff-18c4abf20356408efc28d1d949b45f1b8f479898f849eb0af84e538ebd9322f4R282) used in [Aztec](https://github.com/wordpress-mobile/AztecEditor-Android/blob/trunk/aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecHeadingSpan.kt#L95-L100) to [subtract it](https://github.com/WordPress/gutenberg/pull/38205/files#diff-18c4abf20356408efc28d1d949b45f1b8f479898f849eb0af84e538ebd9322f4R331) from the font size before setting the value.
- It adds `isHeadingBlock` to check if the current tag is from a Heading block, e.g `h1`, `h2`, etc.
- Whenever `setBlockType` is called, it triggets `setFontSize`, for cases when the Heading block was already created but its level changes e.g from `h1` to `h2`.

This also limits the minimum value that can be set for line-height to `1`, lower values break on Android where the block doesn't show the correct height.

### Test case 1

- Add a Paragraph block
- Type in some text
- Open the block settings
- Change the font size
- **Expect** to see the text with the selected new size

### Test case 2

- Add a Heading block
- Type in some text
- Change through different levels, `h1`, `h3`, `h4`
- **Expect** to see the block change accordingly

### Test case 3

- Add a Heading block
- Set the `H1` level
- Type in a long text so there are multiple lines
- **Expect** the text to show correctly without overlapping
- Open the block settings
- Change the font size
- **Expect** to see the text with the new font size
- Change to the `H4` level
- **Expect** the text to keep the same font size selected before
- Open the block settings
- Select the `Default` font size
- **Expect** to see the font size go back to the selected heading level default size

### Test case 1

- Open the app
- Add a Paragraph block
- Type in several lines of text
- Open the block settings
- Change the line-height
- **Expect** to see the line height to change

### Test case 2

- Add a Heading block
- Type in several lines of text
- Open the block settings
- Change the line-height
- **Expect** to see the line height to change
- Change the heading level to `H1`
- **Expect** the font size value to change and the line-height to update accordingly to the new size

### Test case 3

- Add a Heading block
- Type in several lines of text
- Open the block settings
- **Check** that you **can't select** a line-height value lower than `1`

### Test case 1

- Open the app
- Add a Paragraph block
- Type in some text
- Open the block settings
- **Expect** to **not** see the font size picker

### Test case 2

- Open the app
- Add a Heading block
- Type in some text
- Change to `H1`
- **Expect** the font size to change
- Change to `H3`
- **Expect** the font size to change

### Test case 1

- Open the app
- Add a Heading block
- Type in some text
- Open the block settings
- **Expect** to **not** see the line height picker

## Font size (block-based themes: Calvin, Quadrat, etc.)

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd></kbd> | <kbd></kbd> | <kbd></kbd>

## Line-height (block-based themes: Calvin, Quadrat, etc.)

Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd></kbd> | <kbd></kbd> | <kbd></kbd>

## Font size (standard theme: Seedlet)

Test case 1 | Test case 2
-|-
<kbd></kbd> | <kbd></kbd>

## Line-height (standard theme: Seedlet)

Test case 1
<kbd></kbd>

## Types of changes
New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [x] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java [hunk 1]
```diff
@@ -267,6 +270,34 @@ private void updateSelectionIfNeeded(ReactAztecText view, @Nullable ReadableMap
...
+    private boolean isHeadingBlock(ReactAztecText view) {
+        String tag = view.getTagName();
+        final String regex = "h([1-6])";
+        final Pattern pattern = Pattern.compile(regex);
+        final Matcher matcher = pattern.matcher(tag);
+
+        return matcher.find();
+    }
+
+    private float getHeadingScale(String scale) {
```
Comment (antonis) [inline packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:282]: This is more of a question than a change request. What is the source of these numbers and are they used in other parts of the code?

**Review by antonis — APPROVED** [review]
Great work @geriux 👍 
I tested the changes on a Pixel 5 (Android 12) and an iPhone SE 2020 (iOS 14.7.1) and everything worked as expected. The code changes also LGTM 🎉 
Thank you for wrangling this 🙇

File: packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java (hunk 1 above)
Comment (geriux) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:282]: Sure! These values were taken from [here](https://github.com/wordpress-mobile/AztecEditor-Android/blob/trunk/aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecHeadingSpan.kt#L94-L100). I tried to move them to the [Constants](https://github.com/wordpress-mobile/AztecEditor-Android/blob/trunk/aztec/src/main/kotlin/org/wordpress/aztec/Constants.kt) file using something like:

```kotlin
object Constants {
...
    val SCALE_H1 = 1.73f
    val SCALE_H2 = 1.32f
    val SCALE_H3 = 1.02f
    val SCALE_H4 = 0.87f
    val SCALE_H5 = 0.72f
    val SCALE_H6 = 0.60f
}
```

But to get them on `ReactAztecManager` I only found a way to get them using `Constants.INSTANCE.getSCALE_H1();` and apparently, it's not good for performance 🤔 

Maybe it'd be a good idea to reference where those values are from within `getHeadingScale`, what do you think? Thanks!

Comment (antonis) [reply packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:282]: > to reference where those values are from within `getHeadingScale`

I think that's a good idea for future reference 👍

**Pull request merged**